midifile.c

```diff
@@ -17,8 +17,52 @@
 #include "plugout.h"
 #include "midifile.h"
 
-#define TRACK_LENGTH_OFFSET 18
-#define TRACK_START_OFFSET 22
+/*
+ * The hardware emulation runs at 4194304 Hz and provides an exact
+ * cycle counter.  Because ~4 MHz are outside of the possible MIDI
+ * time resolution, we introduce an arbitrary lower resolution of 256
+ * Hz by shifting the cycle counter 14 bits to the right.
+ *
+ * This lower resolution (#FIXME: which needs a nice name) is
+ *
+ *  4194304 Hz / 2^14 == 1/256 Hz == 3906 ms
+ */
+static const uint8_t CYCLE_RESOLUTION_BIT_SHIFT = 14;
+
+/*
+ * Now we need to make a single MIDI tick match the 3906ms from above.
+ * 
+ * In MIDI, the Time Division use to define the length of a MIDI tick.
+ * It is a 16 bit value with two possible calculation modes:
+ * Top bit == 0 selects PPQ (Pulses per quarter note)
+ * Top bit == 1 selects Frames per Second
+ *
+ * When using PPQ (pulses per quarter note) mode, the length of a
+ * single MIDI tick is calculated as:
+ *
+ *  1 tick = TEMPO / TIME_DIVISION
+ *
+ * TEMPO is measured in ms/beat (1 beat == 1 quarter note)
+ * and can be set by the tempo command.
+ *
+ * A nicely matching pair of values would be:
+ *
+ *  TEMPO         = 499968 ms / beat
```

Comment:
I think the MIDI time unit is also specified in microseconds. (499968µs/beat would amount to about 120bpm so it looks plausible)